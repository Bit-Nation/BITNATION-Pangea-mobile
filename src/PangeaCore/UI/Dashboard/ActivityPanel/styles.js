// @flow

import { MediaQueryStyleSheet } from 'react-native-responsive';

import GlobalStyles from 'pangea-common-reactnative/styles';
import Colors from 'pangea-common-reactnative/styles/colors';

const styles = MediaQueryStyleSheet.create({
  ...GlobalStyles,
  flex: { flex: 1 },
  panelViewTitle: {
    ...GlobalStyles.panelViewTitle,
    color: Colors.BitnationHighlightColor,
  },
});
export default styles;
