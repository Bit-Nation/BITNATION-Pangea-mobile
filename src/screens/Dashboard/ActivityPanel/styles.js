import { MediaQueryStyleSheet } from 'react-native-responsive';
import { Dimensions } from 'react-native';

import GlobalStyles from '../../../global/Styles';
import Colors from '../../../global/colors';

const styles = MediaQueryStyleSheet.create(
  {
    ...GlobalStyles,
    flex: { flex: 1 },

    panelViewTitle: {
      ...GlobalStyles.panelViewTitle,
      color: Colors.BitnationHighlightColor,
    },

  });
export default styles;