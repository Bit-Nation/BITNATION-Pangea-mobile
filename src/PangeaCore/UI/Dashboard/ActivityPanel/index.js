// @flow

import React from 'react';
import {
  FlatList,
  Text,
  View,
} from 'react-native';

import styles from './styles';
import PanelView from '../../../components/common/PanelView';
import i18n from '../../../global/i18n';
import Button from '../../../components/common/Button';
import type { ActivityLogMessage } from '../../../types/ActivityLogMessage';

type Props = {
  /**
   * @desc Array of messages objects to display.
   */
  +messages: Array<ActivityLogMessage>,
  /**
   * @desc Style to be applied to root view.
   */
  style: any,
};

type TestingModeProps = {
  /**
   * @desc State of testing mode. For testing only.
   */
  testingMode: {
    +isActive: bool,
  },
  /**
   * @desc Callback to add dummy message. For testing only.
   */
  onAddDummyMessage: () => void,
}

/**
 * @desc Component to render nation panel on dashboard
 * @return {React.Component} A component.
 */
const ActivityPanel = ({
  style, messages, testingMode, onAddDummyMessage,
}: Props & TestingModeProps) => ((
  <View style={style}>
    <PanelView
      style={styles.gridPanelView}
      titleStyle={styles.panelViewTitle}
      childrenContainerStyle={styles.flex}
      title={i18n.t('screens.dashboard.activityPanel.title')}
    >
      <FlatList
        renderItem={(item: { item: ActivityLogMessage }) => {
          const message = item.item;
          const messageText = message.interpret ? i18n.t(`activityLog.${message.msg}`, JSON.parse(message.params)) : message.msg;
          return (<Text style={styles.listItemText}>{messageText}</Text>);
        }}
        keyExtractor={(item: ActivityLogMessage) => item.id.toString()}
        data={messages}
        style={styles.flex}
      />
      {
        testingMode.isActive === true &&
        <Button title='Add dummy log' onPress={onAddDummyMessage} />
      }

    </PanelView>
  </View>
));

ActivityPanel.defaultProps = {
  messages: [],
};

export default ActivityPanel;
