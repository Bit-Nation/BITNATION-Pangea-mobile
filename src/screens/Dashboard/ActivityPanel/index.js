import React, { Component } from 'react';
import {
  FlatList,
  Text,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import PanelView from '../../../components/common/PanelView';
import i18n from '../../../global/i18n';
import Button from '../../../components/common/Button';

/**
 * @desc Component to render nation panel on dashboard
 * @type React.Component
 */
export default class ActivityPanel extends Component {

  render() {
    const { style } = this.props;

    return (
      <View style={style}>
        <PanelView style={styles.gridPanelView}
                   titleStyle={styles.panelViewTitle}
                   childrenContainerStyle={styles.flex}
                   title={i18n.t('screens.dashboard.activityPanel.title')}>
          <FlatList
            renderItem={(item) => {
              const message = item.item;
              const messageText = message.interpret ? i18n.t(`activityLog.${message.msg}`, JSON.parse(message.params)) : message.msg;
              return (<Text style={styles.listItemText}>{messageText}</Text>);
            }}
            keyExtractor={(item) => item.id}
            data={this.props.messages}
            style={styles.flex}
          />
          {
            this.props.testingMode.isActive &&
            <Button title='Add dummy log' onPress={this.props.onAddDummyMessage}/>
          }

        </PanelView>
      </View>
    );
  }

}

ActivityPanel.propTypes = {
  /**
   * @desc Array of messages objects to display
   */
  messages: PropTypes.array,
  /**
   * @desc State of testing mode. For testing only.
   */
  testingMode: PropTypes.object,
  /**
   * @desc Callback to add dummy message. For testing only.
   */
  onAddDummyMessage: PropTypes.func,
};

ActivityPanel.defaultProps = {
  messages: [],
};
