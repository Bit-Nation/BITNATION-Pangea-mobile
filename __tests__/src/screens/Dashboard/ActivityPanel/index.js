import React from 'react';
import { shallow } from 'enzyme';

import ActivityPanel from '../../../../../src/screens/Dashboard/ActivityPanel';

test('ActivityPanel renders correctly', () => {
  const wrapper = shallow((
    <ActivityPanel
      messages={[]}
      testingMode={{ isActive: true }}
      onAddDummyMessage={jest.fn()}
    />
  ));
  expect(wrapper).toMatchSnapshot();
});
