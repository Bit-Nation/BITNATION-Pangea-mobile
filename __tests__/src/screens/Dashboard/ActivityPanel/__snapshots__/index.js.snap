// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityPanel renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivityPanel
    messages={Array []}
    onAddDummyMessage={[MockFunction]}
    testingMode={
      Object {
        "isActive": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <PanelView
        body={undefined}
        buttonTitle="Button"
        childrenContainerStyle={
          Object {
            "flex": 1,
          }
        }
        icon={undefined}
        onButtonClick={null}
        renderAdditionalInfo={null}
        renderBottom={null}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "margin": 4,
            "overflow": "hidden",
            "padding": 16,
          }
        }
        title="Activity"
        titleStyle={
          Object {
            "color": "#F5A623",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontWeight": "bold",
            "letterSpacing": 0,
            "lineHeight": 28,
            "textAlign": "left",
          }
        }
      >
        <FlatList
          data={Array []}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          style={
            Object {
              "flex": 1,
            }
          }
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
        <Button
          enabled={true}
          onPress={[MockFunction]}
          style={undefined}
          title="Add dummy log"
        />
      </PanelView>,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "body": undefined,
        "buttonTitle": "Button",
        "children": Array [
          <FlatList
            data={Array []}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "flex": 1,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
          <Button
            enabled={true}
            onPress={[MockFunction]}
            style={undefined}
            title="Add dummy log"
          />,
        ],
        "childrenContainerStyle": Object {
          "flex": 1,
        },
        "icon": undefined,
        "onButtonClick": null,
        "renderAdditionalInfo": null,
        "renderBottom": null,
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "margin": 4,
          "overflow": "hidden",
          "padding": 16,
        },
        "title": "Activity",
        "titleStyle": Object {
          "color": "#F5A623",
          "fontFamily": "Roboto",
          "fontSize": 22,
          "fontWeight": "bold",
          "letterSpacing": 0,
          "lineHeight": 28,
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "style": Object {
              "flex": 1,
            },
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "enabled": true,
            "onPress": [MockFunction],
            "style": undefined,
            "title": "Add dummy log",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <PanelView
          body={undefined}
          buttonTitle="Button"
          childrenContainerStyle={
            Object {
              "flex": 1,
            }
          }
          icon={undefined}
          onButtonClick={null}
          renderAdditionalInfo={null}
          renderBottom={null}
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "rgba(255,255,255,0.2)",
              "borderRadius": 8,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "margin": 4,
              "overflow": "hidden",
              "padding": 16,
            }
          }
          title="Activity"
          titleStyle={
            Object {
              "color": "#F5A623",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            }
          }
        >
          <FlatList
            data={Array []}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "flex": 1,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
          <Button
            enabled={true}
            onPress={[MockFunction]}
            style={undefined}
            title="Add dummy log"
          />
        </PanelView>,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "body": undefined,
          "buttonTitle": "Button",
          "children": Array [
            <FlatList
              data={Array []}
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              style={
                Object {
                  "flex": 1,
                }
              }
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
            <Button
              enabled={true}
              onPress={[MockFunction]}
              style={undefined}
              title="Add dummy log"
            />,
          ],
          "childrenContainerStyle": Object {
            "flex": 1,
          },
          "icon": undefined,
          "onButtonClick": null,
          "renderAdditionalInfo": null,
          "renderBottom": null,
          "style": Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "margin": 4,
            "overflow": "hidden",
            "padding": 16,
          },
          "title": "Activity",
          "titleStyle": Object {
            "color": "#F5A623",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontWeight": "bold",
            "letterSpacing": 0,
            "lineHeight": 28,
            "textAlign": "left",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [],
              "disableVirtualization": false,
              "horizontal": false,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "style": Object {
                "flex": 1,
              },
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "enabled": true,
              "onPress": [MockFunction],
              "style": undefined,
              "title": "Add dummy log",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
